tests/system-tests/internal/apiobjectshelper/apiobjectshelper.go

```diff
@@ -81,3 +91,280 @@ func VerifyOperatorDeployment(apiClient *clients.Settings,
 
 	return nil
 }
+
+// DeleteDeployment deletes the deployment and verifies it and all related pods were removed.
+func DeleteDeployment(
+	apiClient *clients.Settings,
+	deploymentName, nsName, deploymentLabel string) error {
+	glog.V(100).Infof("Removing test deployment %q from %q ns", deploymentName, nsName)
+
+	if snifferDeployment, err := deployment.Pull(apiClient, deploymentName, nsName); err == nil {
+		glog.V(100).Infof("Deleting deployment %q from %q namespace", deploymentName, nsName)
+
+		err = snifferDeployment.DeleteAndWait(300 * time.Second)
+
+		if err != nil {
+			return fmt.Errorf("failed to delete deployment %q from %q namespace: %w", deploymentName, nsName, err)
```

Comment:
nit: worth logging that error's happened during deletion as well as a successful deletion